source/MaterialXGenShader/ShaderTranslator.cpp

```diff
@@ -32,13 +32,22 @@ void ShaderTranslator::connectTranslationInputs(NodePtr shader, NodeDefPtr trans
             {
                 InputPtr input = _translationNode->addInput(shaderInput->getName(), shaderInput->getType());
                 input->setConnectedNode(_graph->getNode(output->getNodeName()));
-
+                if (shaderInput->getColorSpace() != EMPTY_STRING) {
+                  input->setColorSpace(shaderInput->getColorSpace());
+                }
                 _graph->removeOutput(output->getName());
             }
             else if (shaderInput->getValueString() != EMPTY_STRING)
             { 
                 InputPtr input = _translationNode->addInput(shaderInput->getName(), shaderInput->getType());
                 input->setValueString(shaderInput->getValueString());
+                if (shaderInput->getColorSpace() != EMPTY_STRING) {
+                    input->setColorSpace(shaderInput->getColorSpace());
+                }
+                if (shaderInput->getUnit() != EMPTY_STRING) {
+                    input->setUnit(shaderInput->getUnit());
+                    input->setUnitType(shaderInput->getUnitType());
```

Comment:
If getUnitType is empty, is it default or empty?

Reply:
Empty.